nav2_map_server/include/nav2_map_server/logging.hpp

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2020 Samsung R&D Institute Russia
+// Copyright (c) 2017 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Allows to using logging in static methods where rclcpp::Logger
+// object is not available.
+
+#ifndef NAV2_MAP_SERVER__LOGGING_HPP_
+#define NAV2_MAP_SERVER__LOGGING_HPP_
+
+#include "rclcpp/rclcpp.hpp"
+
+#define MAP_SERVER_INFO(logger_name, ...) \
```

Comment:
Why not just use `RCLCPP_INFO`, `RCLCPP_WARN` etc?

Reply:
Hardcoding the "map_server" in each of your wrapped calls makes it so if you name the node something different, or have multiple instances, they appear on screen in that hardcoded string. This is why we use the `get_logger()` so its changing by the node name passed in the launch file

Reply:
RCLCPP_INFO/WARN/etc... macroses are using `logger` as a first argument. From inside static class methods (like `saveMapToFile()`) I cannot reference on map_saver class object and therefore on its logger. So, I've decided to create our own `MAP_SERVER_INFO()/WARN()/etc...` macroses where the logger name will be passed a string. However, this does not resolve well the situation you've discribed when we have multiple instances of node each one with different names.

Here appears more "big" question: do we need to have static API methods for dynamic library or is it better to switch to common non-static public methods?
For the reference: there are three static API methods in this patch:
```
* static bool saveMapToFile(
    const nav_msgs::msg::OccupancyGrid & map,
    SaveParameters & save_parameters);
* static void loadMapFromFile(
    const LoadParameters & load_parameters,
    nav_msgs::msg::OccupancyGrid & map);
* static bool loadMapFromYaml(
    const std::string & yaml_file,
    nav_msgs::msg::OccupancyGrid & map);
```
The initial idea was to use these static API methods to easy call them directly from linked library `libmap_server_core.so/.dll` without having local copy of `map_saver`/`map_server` object.
On other hand, to solve multiple-name issue, it might be required to add one more parameter to static API calls which seems to be sloppy.

Reply:
Maybe we don't need logging in the static methods, you throw exceptions so if there's something that handles the exceptions and then prints `e.what()` to screen, you shouldn't need to log inside of the static methods themselves.

I don't really call personally if its static or non static. Its static right now so that multiple things can be doing file IO at the same time since they share an instance. Static shouldn't be used in this context in my opinion to call a function outside an instance of an object. If that was your goal, I think you should be creating a specific `MapIO` object that contains these methods that can then be used by other classes, but not my calling static methods from a non-instantiated object. That seems abusive to intent. 

The typical way we make ROS programs is to have a non-ROS library with ROS-wrapped applications. It seems we're missing a bit of that here. Having some basic IO library that the map saver and map loader uses, then an application. That way different IO can be swapped out over time without rewriting the program and separating the concerns.

Reply:
Applied in 84cff7d: Added new `MapIO` (`mapio.hpp`/`mapio.cpp`) library